### PR TITLE
#2151 check the jar key for both creation request and update request

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2151)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala [hunk 1]
```diff
@@ -302,6 +302,8 @@ object StreamCollie {
       */
     def jarInfo(jarInfo: FileInfo): ClusterCreator = {
       request.jarInfo(jarInfo)
+      // Since the jarKey is required in API, we fill the "required" field by jarInfo
+      request.jarKey(ObjectKey.of(jarInfo.group, jarInfo.name))
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala]: request.jarKey(jarInfo.key)

**Comment by saivirtue** [comment]
> Could you also revise the docs?

Sure, Address new subtasks in #2151

**Comment by saivirtue** [comment]
retry manager-api

**Comment by saivirtue** [comment]
The failed test is related to #2660 and should be fixed. Merge this issue later.

**Pull request merged**